src/Dijkstra/AST/Examples/PaperIntro.agda

```diff
@@ -17,10 +17,10 @@ open import Relation.Binary.PropositionalEquality
 
 -- The example in the paper
 prog : (St → Maybe Wr) → RWSAST Unit
-prog f = pass inner
+prog g = pass inner
```

Comment:
I will change the paper to use `g` as well.  This is not a gratuitous change, but one carefully chosen to avoid confusion.  The function is passed to `gets`, which is syntactic sugar for `RWSgets`, which is defined in terms of a function `g`.  So I'm making that consistent, in particular to avoid confusion with `f` used in places like `bindPT`, which is entirely different and unrelated, but is seen by someone navigating the example.  While writing this, I noticed that the syntactic sugar actually uses `f`, so I've changed that as well, and similarly for functions used in the syntactic sugar for other `RWSCmd`s.  Thanks as always for provoking me!

Reply:
Are you saying that the variables used inside the implementation of `bindPT` and/or `RWSgets` are already using `f`?  So you changed to avoid confusion?

If that is true
- avoiding confusion is a good thing
- changing "user code" (which is the proof) to avoid clashes with names in "library" code (which is what our Dijkstra is) is not a good way to go.   That said, trying to use variable names in our library that won't clash with user code is difficult.

In this case I suggest leaving the paper as `f`, updating our library code to not use `f` and reverting the tutorial example back to `f`.   If that is too much, then go with `g` - but examples in papers usually use `f` first (and `g` later if there are multiple functions).

Reply:
No, I said that the function in the definition of `RWgets` is `g` and the one in `bindPT` is `f`.  So calling the function in the example that is passed to `gets` `f` leads to unnecessary confusion when one is trying to understand the example and how it interacts with the framework.

Reply:
And what I was saying is we should change the variable names in `RWSgets` and `bindPT` so the user can "have" `f` and `g`.   It's a useability issue - not just for the paper.

Reply:
Discussed and I think we're on the same page now.